packages/ssx-sdk/src/ssx.ts

```diff
@@ -107,15 +113,93 @@ export class SSX {
   public storage: IStorage;
 
   constructor(private config: SSXConfig = SSX_DEFAULT_CONFIG) {
-    // TODO: initialize these based on the config
+    // TODO: pull out config validation into separate function
+    // TODO: pull out userAuthorization config
     this.userAuthorization = new UserAuthorization(config);
-    // get encryption config from config.modules.encryption
-    // determine which encryption module to use
-    // if encryption module is false, don't initialize encryption or dependent modules
-    this.encryption = new SignatureEncryption({}, this.userAuthorization);
-    this.dataVault = new BrowserDataVault({}, this.encryption);
-    this.credential = new Credential({}, this.dataVault);
-    this.storage = new BrowserStorage({});
+
+    // credential module disabled for now
+    // this.credential = new Credential({}, this.dataVault);
+
+    // initialize encryption module
+    // assume encryption module is **enabled** if config.encryption is not defined
+    const encryptionConfig =
+      config?.modules?.encryption === undefined
+        ? true
+        : config.modules.encryption;
+    if (encryptionConfig !== false) {
+      if (typeof encryptionConfig === 'object') {
+        // determine which encryption module to use
+        // initialize encryption with the provided config
+        if (encryptionConfig.module === 'SignatureEncryption') {
+          this.encryption = new SignatureEncryption(
+            encryptionConfig,
+            this.userAuthorization
+          );
+        } else {
+          throw new Error(
+            `Encryption module ${encryptionConfig.module} not supported`
+          );
+        }
+        // } else if (typeof encryptionConfig === 'function') {
+        //   // Initialize encryption with the return from the provided function
+        //   this.encryption = encryptionConfig();
+      } else {
+        // encryption == true or undefined
+        // Initialize encryption with default config when no other condition is met
+        this.encryption = new SignatureEncryption({}, this.userAuthorization);
+      }
+      // IEncryption does not yet extend ISSXExtension
+      // this.extend(this.encryption);
```

Comment:
When we add LitEncryption, we can add this back in and have IEncryption extend ISSXExtension